scripts/buildorder.py

```diff
@@ -347,19 +333,64 @@ def generate_target_buildorder(target_path, pkgs_map, fast_build_mode):
         package.deps.difference_update([subpkg.name for subpkg in package.subpkgs])
     return package.recursive_dependencies(pkgs_map)
 
+def get_list_cyclic_dependencies(pkgs_map, index=None, checked=None, pkgname=None, return_list=False):
+    "Find and print (or return a list of) circular dependencies for all packages or for one specified package."
+    # is_root - indicates that the function was run for the first time.
+    # Here such functions are called "root function".
+    is_root = index == None
+    if is_root:
+        checked = []
+        index = []
+    result = []
+
+    # Start search
+    if is_root:
+        if pkgname:
+            range_pkgs = [pkgname]
+        else:
+            range_pkgs = pkgs_map.keys()
+    else:
+        range_pkgs = pkgs_map[index[-1]].deps
+    for pkg in range_pkgs:
+        if pkg in checked:
+            continue
+        index.append(pkg)
+        if index.count(pkg) == 2:
+            result.append(index.copy() if pkgname else index[index.index(pkg)::])
+        else:
+            result += get_list_cyclic_dependencies(pkgs_map, index, checked, pkgname)
+        del index[-1]
+        checked.append(pkg)
+
+    # return result
+    if is_root and not return_list:
+        # If launched in a root function and the `return_list` variable is set to False,
+        # then the result will be printed
+        if len(result) == 0:
+            print("No cyclic dependencies were found")
+        else:
+            print(f"Found {len(result)} cyclic dependencies:")
+            for cycle in result:
+                print("- "+" -> ".join(cycle))
+    else:
+        return result
+
 def main():
     "Generate the build order either for all packages or a specific one."
     import argparse
 
     parser = argparse.ArgumentParser(description='Generate order in which to build dependencies for a package. Generates')
     parser.add_argument('-i', default=False, action='store_true',
                         help='Generate dependency list for fast-build mode. This includes subpackages in output since these can be downloaded.')
+    parser.add_argument('-l', default=False, action='store_true',
```

Comment:
> [!NOTE]
> Isn't the current behavior to output circular dependencies without the need for any flag?

Reply:
> Isn't the current behavior to output circular dependencies without the need for any flag?

Previously, the output of cyclic dependencies occurred in the `generate_full_buildorder` function as an error. Now, since the `generate_full_buildorder` function can work with cyclic dependencies, and this function is launched without entering options (in `buildorder.py`), it was more logical to create a new flag for outputting cyclic dependencies. In addition, with the new flag you can specify a package, which will reduce the search overview and find out whether the package has cyclic dependencies on itself or on another package.